src/v/kafka/server/handlers/handler_probe.h

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "kafka/protocol/types.h"
+#include "ssx/metrics.h"
+
+namespace kafka {
+
+class handler_probe {
+public:
+    explicit handler_probe(api_key);
+    void setup_metrics();
+
+    void sample_in_progress();
+    void request_completed() {
+        sample_in_progress();
+
+        _requests_completed++;
+        _requests_in_progress--;
+    }
+    void request_errored() { _requests_errored++; }
+    void request_started() {
+        sample_in_progress();
+
+        _requests_in_progress++;
+    }
+
+    void add_bytes_received(size_t bytes) { _bytes_received += bytes; }
+
+    void add_bytes_sent(size_t bytes) { _bytes_sent += bytes; }
+
+private:
+    ss::metrics::metric_groups _metrics;
+    api_key _key;
+
+    uint64_t _requests_completed{0};
+    uint64_t _requests_errored{0};
+
+    uint64_t _requests_in_progress{0};
+    uint64_t _requests_in_progress_every_ns{0};
+
+    ss::lowres_clock::time_point _last_recorded_in_progress;
```

Comment:
This value is never initialized, so I guess it has value "epoch", does that DTRT the first time it is used? Probably not: the running sum should probably only be updated on the 2nd and subsequent sample points.

Reply:
Oh yeah current time seems like it would work too: the first sample point will just end up adding 0 to the integral since req_in_p is 0, right? Perhaps even the current thing is fine since it's always multiplied by 0?

Reply:
Yeah, the only reason why the behavior right now is correct is because `req_in_p` will always be zero in the initial function call. Better to just explicitly initialize `_last_recorded_in_progress` to something sane though in case someone down the line changes something and `req_in_p` is no longer zero initially.

Reply:
SG, this is still TODO, right?

Reply:
This has been implemented. Let me know if it looks good to you.

Reply:
I don't think setting `now()` is much better than "epoch()" (no request is starting at now() either) but I think it's fine because the * 0 means none of this matters anyway.